src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/PublicationServiceTests.cs

```diff
@@ -285,9 +290,75 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
 
                 Assert.Equal(publication.Id, publicationViewModel.Id);
                 Assert.False(publicationViewModel.IsSuperseded);
+                Assert.Null(publicationViewModel.SupersededBy);
             }
         }
 
+        [Fact]
+        public async Task Get_IsSuperseded_SupersedingPublicationBuildsSupersededViewModel()
+        {
+            var releaseId = Guid.NewGuid();
+
+            var publication = new Publication
+            {
+                Title = "Publication A",
+                Slug = "publication-a",
+                LatestPublishedReleaseId = releaseId,
+                SupersededBy = new Publication
+                {
+                    Id = Guid.NewGuid(),
+                    Title = "Publication B",
+                    Slug = "publication-b",
+                    LatestPublishedReleaseId = Guid.NewGuid()
+                }
+            };
+            
+            var themes = new List<Theme>
+            {
+                new()
+                {
+                    Title = "Theme A",
+                    Summary = "Theme A summary",
+                    Topics = new List<Topic>
+                    {
+                        new()
+                        {
+                            Title = "Topic A",
+                            Publications = ListOf(publication)
+                        },
+                    },
+                },
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryContentDbContext(contextId))
+            {
+                await context.Themes.AddRangeAsync(themes);
+                await context.SaveChangesAsync();
+            }
+
+            await using (var context = InMemoryContentDbContext(contextId))
+            {
+                var service = SetupPublicationService(context);
+
+                var publicationTree = await service.GetPublicationTree();
+
+                var testTopicPublications = publicationTree[0].Topics[0].Publications;
+                
+                Assert.Single(testTopicPublications);
+                
+                var testTopicPublication = testTopicPublications[0];
+                
+                Assert.True(testTopicPublication.IsSuperseded);
+                
+                Assert.NotNull(testTopicPublication.SupersededBy);
+                Assert.Equal(publication.SupersededBy!.Id, testTopicPublication.SupersededBy!.Id);
```

Comment:
You don't need the first null-forgiving `!` operator in `publication.SupersededBy!.Id` because it's been declared above with a `SupersededBy` so it's known to be not-null here.